src/mlpack/methods/ann/layer/batch_norm.hpp

```diff
@@ -44,39 +47,64 @@ namespace ann /** Artificial Neural Network. */ {
  * }
  * @endcode
  *
- * @tparam InputDataType Type of the input data (arma::colvec, arma::mat,
- *         arma::sp_mat or arma::cube).
- * @tparam OutputDataType Type of the output data (arma::colvec, arma::mat,
- *         arma::sp_mat or arma::cube).
+ * @tparam MatType Matrix representation to accept as input and use for
+ *         computation.
  */
-template <
-  typename InputType = arma::mat,
-  typename OutputType = arma::mat
->
-class BatchNorm : public Layer<InputType, OutputType>
+template <typename MatType = arma::mat>
+class BatchNormType : public Layer<MatType>
 {
  public:
   //! Create the BatchNorm object.
-  BatchNorm();
+  BatchNormType();
```

Comment:
We should document that this will default `maxAxis` to `2` (or effectively `1` or `0` if there are fewer inputndimensions than `3`).